data/transactions/logic/sourcemap_test.go

```diff
@@ -14,15 +14,42 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
 
-package main
+package logic
 
 import (
+	"strings"
 	"testing"
 
 	"github.com/algorand/go-algorand/test/partitiontest"
 	"github.com/stretchr/testify/require"
 )
 
+func TestGetSourceMap(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	a := require.New(t)
+
+	sourceNames := []string{"test.teal"}
+	offsetToLine := map[int]int{
+		1: 1,
+		2: 2,
+		3: 5,
+	}
+	actualSourceMap := GetSourceMap(sourceNames, offsetToLine)
+
+	a.Equal(sourceMapVersion, actualSourceMap.Version)
+	a.Equal("", actualSourceMap.File)
+	a.Equal(sourceNames, actualSourceMap.Sources)
+	a.Equal([]string{}, actualSourceMap.Names)
+
+	// Check encoding for each line.
+	splitMapping := strings.Split(actualSourceMap.Mapping, ";")
+	for pc := range splitMapping {
+		if line, ok := offsetToLine[pc]; ok {
```

Comment:
@algochoi Minor:  What are your thoughts on testing `!ok` case?

Unless I've missed some intuition:  When `ok` is false, it feels prudent to confirm `splitMapping[pc]` encodes empty value.

If you're on-board with the idea, I'd consider writing a specific sub-test or isolated test.  It's possible someone changes the existing test such that the condition won't be tested.

Reply:
Good point, I'll add an `else` statement to check that pc values that do not correspond to a line in the source matches an empty string in the encoded mapping.

Reply:
@algochoi Thanks for taking in the feedback in https://github.com/algorand/go-algorand/commit/6645cb5eebaecb7476ff4849ad6573d25f8c74c5.  Resolving.